bn_mp_set_double.c

```diff
@@ -41,19 +54,119 @@ int mp_set_double(mp_int *a, double b)
       return res;
    }
 
-   if (((cast.bits >> 63) != 0ULL) && !IS_ZERO(a)) {
-      a->sign = MP_NEG;
+   if (((cast.bits >> PAYLOAD) != 0ULL) && (mp_iszero(a) == MP_NO)) {
+      SIGN(a) = MP_NEG;
    }
 
    return MP_OKAY;
 }
+
+#else
+static double s_math_h_less_frexp(double x, int *exp)
```

Comment:
Is s_ the prefix used for static private functions? I guess I am fine with that.

Reply:
Yes, it is. Didn't know it either but got hit over my head for not using it.

Reply:
I've also seen s_mp_*, fast_s_mp_*, bn_reverse, mp_s_*, KARATSUBA_*, TOOM_* in a list of symbols. Maybe we should clean that up at some point.